src/value/owned.rs

```diff
@@ -276,6 +276,9 @@ impl PartialEq for OwnedStruct {
                 return false;
             }
         }
+        if other.no_text_fields.ne(&self.no_text_fields) {
+            return false;
+        }
```

Comment:
I don't think this is right--you either need to generate the nested loop similar to what you have above O(N**2), or materialize one side into a `BTreeSet`/`HashSet` and do the probe on that index at a cost of O(N lg N)/O(N).

```rust
self.no_text_fields.iter().all(|(my_k, my_v)| {
    other
        .no_text_fields
        .iter()
        .find(|(other_k, other_v)| my_k == other_k && my_v == other_v)
        .is_some()
})
```